src/Middleware/RequestBodyParserMiddleware.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace React\Http\Middleware;
+
+use Psr\Http\Message\ServerRequestInterface;
+
+final class RequestBodyParserMiddleware
```

Comment:
I would like to not see final in this case. There are literally a metric ton of different request body formats and this is rather limiting (from the perspective of a http framework). XML, Json, form-urlencoded are only the most popular however there are many custom encodings as well.
For framework usage it would be nice to be able extend.

Also it's possible to have a mixture of encoding types inside an application (I know it sounds weird, but it happens... a lot).

Reply:
@geggleto going to provide a third party middleware that allows this as I also have a use for it 😄

Reply:
See #225 :-)

Reply:
@geggleto We've decided to provide only some core middleware which are required to build a standard PHP/PSR-7 request. Everything else can be easily provided third-party middleware packages. See #221 for reference.

Reply:
https://github.com/WyriHaximus/reactphp-http-middleware-custom-request-body-parsers :sunglasses: